x/staking/side_handler_test.go

```diff
@@ -0,0 +1,1460 @@
+	address := sdk.AccAddress(privateKey.PubKey().Address().Bytes())
+	commonAddr := common.BytesToAddress(address.Bytes())
+
+	chainParams := initApp.ChainKeeper.GetParams(ctx)
+	blockNumber := big.NewInt(10)
+	nonce := big.NewInt(3)
+
+	suite.Run("Success", func() {
+		suite.contractCaller = mocks.IContractCaller{}
+		txReceipt := &ethTypes.Receipt{
+			BlockNumber: blockNumber,
+		}
+
+		msgValJoin := types.NewMsgValidatorJoin(
+			address,
+			validatorId.Uint64(),
+			uint64(1),
+			sdk.NewInt(1000000000000000000),
+			pubkey,
+			txHash,
+			logIndex,
+			blockNumber.Uint64(),
+			nonce.Uint64(),
+		)
+
+		stakingInfoStaked := &stakinginfo.StakinginfoStaked{
+			Signer:          commonAddr,
+			ValidatorId:     new(big.Int).SetUint64(validatorId.Uint64()),
+			ActivationEpoch: big.NewInt(1),
+			Amount:          amount,
+			Total:           big.NewInt(10),
+			SignerPubkey:    pubkey.Bytes()[1:],
+			Nonce:           nonce,
+		}
+
+		suite.contractCaller.On("GetConfirmedTxReceipt", txHash.EthHash(), chainParams.MainchainTxConfirmations).Return(txReceipt, nil)
+
+		addr, err := sdk.AccAddressFromHex(chainParams.ChainParams.StakingInfoAddress)
+		require.NoError(t, err)
+		suite.contractCaller.On("DecodeValidatorJoinEvent", addr, txReceipt, msgValJoin.LogIndex).Return(stakingInfoStaked, nil)
+
+		result := suite.sideHandler(ctx, &msgValJoin)
+		require.Equal(t, uint32(0), result.Code, "Side tx handler should be success")
+		require.Equal(t, tmprototypes.SideTxResultType_YES, result.Result, "Result should be `yes`")
+	})
+
+	suite.Run("No receipt", func() {
+		suite.contractCaller = mocks.IContractCaller{}
+		txReceipt := &ethTypes.Receipt{
+			BlockNumber: blockNumber,
+		}
+
+		msgValJoin := types.NewMsgValidatorJoin(
+			address.Bytes(),
+			validatorId.Uint64(),
+			uint64(1),
+			sdk.NewInt(1000000000000000000),
+			pubkey,
+			txHash,
+			logIndex,
+			blockNumber.Uint64(),
+			nonce.Uint64(),
+		)
+
+		stakingInfoStaked := &stakinginfo.StakinginfoStaked{
+			Signer:          commonAddr,
+			ValidatorId:     new(big.Int).SetUint64(validatorId.Uint64()),
+			Nonce:           nonce,
+			ActivationEpoch: big.NewInt(1),
+			Amount:          amount,
+			Total:           big.NewInt(10),
+			SignerPubkey:    hmCommonTypes.NewPubKey(pubkey.Bytes())[1:],
+		}
+
+		suite.contractCaller.On("GetConfirmedTxReceipt", txHash.EthHash(), chainParams.MainchainTxConfirmations).Return(nil, nil)
+
+		addr, err := sdk.AccAddressFromHex(chainParams.ChainParams.StakingInfoAddress)
+		require.NoError(t, err)
+		suite.contractCaller.On("DecodeValidatorJoinEvent", addr, txReceipt, msgValJoin.LogIndex).Return(stakingInfoStaked, nil)
+
+		result := suite.sideHandler(ctx, &msgValJoin)
+
+		require.NotEqual(t, uint32(0), result.Code, "Side tx handler should Fail")
+		require.Equal(t, tmprototypes.SideTxResultType_SKIP, result.Result, "Result should be `skip`")
+		//require.Equal(t, uint32(common.CodeWaitFrConfirmation), result.Code)
```

Comment:
we can check for result.Code as well, right? The error codes have changed and will be available either in common or sdkerrors.

Reply:
Yes